### PR TITLE
Updated RecastDemo to SDL2

### LINKED ISSUE #135: Upgrade to SDL 2.0
There's currently issues in SDL 1.2 preventing it from linking with RecastDemo when using Visual Studio 2015.  Given this, it's probably time to upgrade to 2.0.  There shouldn't be a whole lot of issues in the process, as RecastDemo mostly just relies on it for input events and creating a window with a gl context.

### PR DESCRIPTION
**Pull request by grahamboree**
The main motivation for this was to allow the project to build in Visual Studio 2015, as SDL 1.2 wasn't being maintained to allow for that.  In addition to the upgrade I changed a bunch of variable names to be less cryptic, though there still could be a lot more cleanup done in main.

Changes:
* Renamed a bunch of variables in main.cpp to be more descriptive.
* Removed unnecessary SDLMain.h and SDLMain.m OSX objective-c class as well as the plist, strings, xib and icns files, which are not needed.
* included cstdio in imguiRenderGL since SDL2 doesn't do it for us like SDL1.2 did.
* Updated premake5 script to support SDL2.  Also updated it to correctly set the debug working directory, as well as copy the SDL2.dll to the target directory in Visual Studio.
* Updated readme with more descriptive & platform-specific demo project setup instructions
* Updated appveyor build script to build vs2015 as well.

Fixes #135

### CODE REVIEW

File: RecastDemo/Source/main.cpp [hunk 1]
```diff
@@ -58,87 +63,92 @@ static SampleItem g_samples[] =
...
-
+static const int g_nsamples = sizeof(g_samples) / sizeof(SampleItem);
 
 int main(int /*argc*/, char** /*argv*/)
 {
 	// Init SDL
 	if (SDL_Init(SDL_INIT_EVERYTHING) != 0)
 	{
-		printf("Could not initialise SDL\n");
+		printf("Could not initialise SDL.  Error: %s", SDL_GetError());
```
Comment (Janiels) [inline RecastDemo/Source/main.cpp]: Newline is missing here.

File: RecastDemo/Source/main.cpp [hunk 2]
```diff
@@ -58,87 +63,92 @@ static SampleItem g_samples[] =
...
-	glEnable(GL_MULTISAMPLE);
+	SDL_SetWindowPosition(window, SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED);
+	SDL_GL_CreateContext(window);
 
-	SDL_WM_SetCaption("Recast Demo", 0);
-	
 	if (!imguiRenderGLInit("DroidSans.ttf"))
 	{
-		printf("Could not init GUI renderer.\n");
+		printf("Could not init GUI renderer.");
```
Comment (Janiels) [inline RecastDemo/Source/main.cpp]: Also here. Was this on purpose? There are a few cases where you have kept them, but here it definitely looks like it was on purpose. :smile:

File: RecastDemo/Source/main.cpp [hunk 3]
```diff
@@ -58,87 +63,92 @@ static SampleItem g_samples[] =
...
+	int errorCode = SDL_CreateWindowAndRenderer(width, height, flags, &window, &renderer);
+
+	if (errorCode != 0 || !window || !renderer)
 	{
-		printf("Could not initialise SDL opengl\n");
+		printf("Could not initialise SDL opengl\nError: %s\n", SDL_GetError());
 		return -1;
 	}
 
-	glEnable(GL_MULTISAMPLE);
```
Comment (Janiels) [inline RecastDemo/Source/main.cpp]: Where has this gone?

**Comment by Janiels** [comment]
I've left a few questions, but otherwise LGTM.

File: RecastDemo/Source/main.cpp (hunk 2 above)
Comment (grahamboree) [inline RecastDemo/Source/main.cpp]: Actually this was an accident.  I had added some extra info here attempting to print SDL_GetError() and then realized that imgui has nothing to do with SDL (duh).  When I removed the additional error logging, I accidentally removed the newline as well. :dizzy_face:

File: RecastDemo/Source/main.cpp (hunk 3 above)
Comment (grahamboree) [inline RecastDemo/Source/main.cpp]: Don't need it.  SDL2 does it for you if you've got `SDL_GL_MULTISAMPLEBUFFERS` set to 1 and `SDL_GL_MULTISAMPLESAMPLES` set to > 0.

**Comment by grahamboree** [comment]
Finally got Travis working... I've been using Ubuntu 15.10 to test this stuff, which has a package `libsdl2-dev` that installs the sdl2 libs and headers in the right places.  Apparently this package was added to the official repositories in 13.10.  Travis is still running 12.04 (!!!) which doesn't have it, so I had to set it up to build from source.  To top it off, Travis doesn't give you sudo access, so you can't just "make install".  You have to install it to a local folder and tell `pkg-config` to find the .pc files there.... I mostly just copied this config:
https://github.com/geckojsc/ooc-sdl2/blob/master/.travis.yml

**Comment by grahamboree** [comment]
Someone want to give this one last look over and merge it in?  Feels weird to merge my own PR... :wink:

**Comment by Janiels** [comment]
Done!

**Comment by hymerman** [comment]
Yep, Travis CI have a very old issue about upgrading their Ubuntu version... ridiculous!

See https://github.com/travis-ci/travis-ci/issues/2046

**Pull request merged**